aiida_pseudo/cli/family.py

```diff
@@ -20,21 +20,28 @@ def cmd_family():
 @cmd_family.command('show')
 @arguments.PSEUDO_POTENTIAL_FAMILY()
 @options.STRINGENCY()
+@options.UNIT(default=None)
 @options_core.RAW()
 @decorators.with_dbenv()
-def cmd_family_show(family, stringency, raw):
+def cmd_family_show(family, stringency, unit, raw):
     """Show details of pseudo potential family."""
     from tabulate import tabulate
 
     if isinstance(family, RecommendedCutoffMixin):
 
-        if stringency is not None and stringency not in family.get_cutoff_stringencies():
-            raise click.BadParameter(f'`{stringency}` is not defined for family `{family}`.', param_hint='stringency')
+        if stringency is not None:
+            try:
+                family.validate_stringency(stringency)
+            except ValueError as exception:
+                raise click.BadParameter(f'{exception}', param_hint='stringency')
 
-        headers = ['Element', 'Pseudo', 'MD5', 'Wavefunction (eV)', 'Charge density (eV)']
+        if unit is None:
+            unit = family.get_cutoffs_unit(stringency)
```

Comment:
This should now fail if no stringency is defined and there is no default defined. We could simply try/catch this `ValueError`. I wonder if there is a better way because this concept will come back more often where we have to separately try/catch the `validate_stringency` and some other function, like `get_cutoffs_unit`. Maybe we should at the very least allow `None` for `stringency` in `validate_stringency` and then if it is `None` _it_ will already try to get the default and raise if it doesn't exist.

Reply:
Indeed, good point and implemented.

Now we still will have quite a bit code duplication like this in the CLI:

```
    try:
        family.validate_stringency(stringency)
    except ValueError as exception:
        raise click.BadParameter(f'{exception}', param_hint='STRINGENCY')

    try:
        family.validate_cutoffs_unit(unit)
    except ValueError as exception:
        raise click.BadParameter(f'{exception}', param_hint='UNIT')
```

Can you pass a click argument to the initialisation of the option descriptor on the same command? Then we could pass `family` and do this validation in the option definition.

Reply:
I guess you can do this from the context. Don't spoil it for me, I'll figure it out. 😬

Reply:
Hmm, I though I could define a `ParamType` and use the `ctx` in e.g. the `convert` method to obtain the `family`. But the `args` and `params` attributes of the `ctx` seem to be empty, so I'm probably going to need to understand a bit more about the inner workings of `click` to figure this out.

Reply:
In the end I only created the `UnitParamType` that has a convert method to check if the provided unit corresponds to the right quantity (energy in this case). The code here is adapted from the `validate_cutoffs_unit` method of the `RecommendedCutoffMixin` and make more general to also check for other quantities. Note that we could also use a decorator from `pint` for this purpose:

https://pint.readthedocs.io/en/stable/wrapping.html#checking-dimensionality

But perhaps our error messages are more succinct and clear as they are.

For the stringency, there is only 1 location where it needs to be validated in the CLI atm, since for the `family cutoffs set` command the stringency should _not_ be validated to be defined on the class already, of course. 🙄

Reply:
That said, if you know how to use the `family` argument in the `covert` method of the type of another option defined on a command, I would love to know. 😁

Reply:
Yeah, this is a known limitation of the design of `click`. Essentially, it is not possible to have validators for parameters that depend on other parameters because the order in which parameters are processed is not well defined. I know that I went down this rabbit-hole a long time ago and came up empty-handed

Reply:
I see. I'll give it another go if we start having a lot of duplicated code because of validation that requires other parameters. 👍